_pytest/python.py

```diff
@@ -1246,6 +1260,11 @@ def __exit__(self, *tp):
         suppress_exception = issubclass(self.excinfo.type, self.expected_exception)
         if sys.version_info[0] == 2 and suppress_exception:
             sys.exc_clear()
+        if self.match_expr:
+            try:
```

Comment:
I wasn't sure how to handle this properly, because `ExceptionInfo.match` raises an AssertionError if no match is found, but we want to call `pytest.fail` in that case...
Is there a better way of handling this?